### PR TITLE
dotCMS/core#17803 Unable to link to page on non-default host for editing

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/api/services/dot-router/dot-router.service.ts [hunk 1]
```diff
@@ -19,6 +19,11 @@ export class DotRouterService {
         };
     }
 
+    get queryParams(): Params {
```
Comment (fmontes) [inline src/app/api/services/dot-router/dot-router.service.ts]: Test

File: src/app/api/services/dot-page-render/dot-page-render.service.ts [hunk 2]
```diff
@@ -25,14 +26,17 @@ export class DotPageRenderService {
      * @returns {Observable<DotPageRender>}
      * @memberof DotPageRenderService
      */
-    get({ viewAs, mode, url }: DotPageRenderOptions): Observable<DotPageRender.Parameters> {
+    get(
```
Comment (fmontes) [inline src/app/api/services/dot-page-render/dot-page-render.service.ts]: Update test.

File: src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts [hunk 3]
```diff
@@ -182,7 +182,9 @@ export class DotPageStateService {
      * @memberof DotPageStateService
      */
     requestPage(options: DotPageRenderOptions): Observable<DotPageRenderState> {
-        return this.dotPageRenderService.get(options).pipe(
+        const { url, ...extraParams } = this.dotRouterService.queryParams;
+        console.log('requestPage', options, extraParams);
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts]: Remove.

File: src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts [hunk 4]
```diff
@@ -51,12 +52,14 @@ export class DotIframeEventsHandler {
         });
     }
 
-    private goToEditPage($event: CustomEvent): void {
+    private goToEditPage($event: CustomEvent<DotCMSEditPageEvent>): void {
+        console.log('iframe', $event.detail.data);
```
Comment (freddyucv) [inline src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts]: remove

File: src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts [hunk 5]
```diff
@@ -51,12 +52,14 @@ export class DotIframeEventsHandler {
...
         this.dotLoadingIndicatorService.show();
-        this.dotRouterService.goToEditPage(
-            $event.detail.data.url,
-            $event.detail.data.content.languageId
-        );
+        this.dotRouterService.goToEditPage({
+            url: $event.detail.data.url,
+            language_id: $event.detail.data.languageId,
+            host_id: $event.detail.data.hostId
```
Comment (freddyucv) [inline src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts]: why no pass just $event.detail.data?

File: src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts [hunk 6]
```diff
@@ -44,11 +45,12 @@ export class DotContentletWrapperComponent implements OnInit {
...
-                'edit-page': (event: CustomEvent<DotCMSEditPageEvent>) => {
-                    this.dotRouterService.goToEditPage(
-                        event.detail.data.url,
-                        event.detail.data.language_id
-                    );
+                'edit-page': ({ detail: { data } }: CustomEvent<DotCMSEditPageEvent>) => {
+                    this.dotRouterService.goToEditPage({
+                        url: data.url,
+                        language_id: data.languageId,
+                        host_id: data.hostId
```
Comment (freddyucv) [inline src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts]: why not just data?

File: src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts (hunk 4 above)
Comment (alfredo-dotcms) [inline src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts]: remove

File: src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts (hunk 5 above)
Comment (fmontes) [reply src/app/view/components/_common/iframe/iframe-porlet-legacy/services/iframe-events-handler.service.ts]: The properties are different

File: src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts (hunk 6 above)
Comment (fmontes) [reply src/app/view/components/dot-contentlet-editor/components/dot-contentlet-wrapper/dot-contentlet-wrapper.component.ts]: the property names are different

**Pull request merged**